homeassistant/components/intellifire/__init__.py

```diff
@@ -39,3 +54,24 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+# Example migration function
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
```

Comment:
Why migrate? You can just raise `ConfigEntryAuthFailed` if you detect that a username is stored in entry.data

Reply:
Most likely because I don't understand migration! :) 

I assumed if I was adding new fields I'd have to have a migration path... is that not the case?

Reply:
Ok this was a great idea I've removed migration ..  HOWEVER ... I'm running into one small issue still. It appears my "reconfigure" box triggered by throwing the `ConfigEntryAuthFailed` error isn't able to correctly pull the `serial` value into 

<img width="303" alt="image" src="https://user-images.githubusercontent.com/6491743/162352358-feb4a326-20c8-49b7-ad71-8a6ac424306d.png">

The actual error is:

> Translation Error: The intl string context variable "serial" was not provided to the string "{serial} ({host})"


<img width="252" alt="image" src="https://user-images.githubusercontent.com/6491743/162352230-39143a6e-a10a-482a-a260-c5b6e395d673.png">
<img width="253" alt="image" src="https://user-images.githubusercontent.com/6491743/162352241-de454c84-00a9-44bc-8839-3b7e1b9fee70.png">

Can you point me to where/how to set this data correctly? I'm a little unsure exactly where / which function is triggering it or where its trying to pull that data from.

Reply:
Triggered by this version of `core.config_entries` ...

<img width="475" alt="image" src="https://user-images.githubusercontent.com/6491743/162765974-6ee412fb-3bc8-4160-bb87-6c803aac025d.png">

Reply:
Even with the translation errors in the UI - The Reauth works correctly and I can correctly update/fix the integration after following the flow